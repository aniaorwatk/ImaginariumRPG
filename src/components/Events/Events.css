.publicationsBox {
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    overflow: hidden;
    box-sizing: border-box;
    padding: 80px 14px 0 14px;
    border-radius: 8px;
    background-color: rgba(59, 61, 81, 0.3);
    filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.07));
}

.publications {
    display: flex;
    flex-wrap: nowrap;
    column-gap: 10px;
    width: 317px;
    overflow-y: hidden;
}

.publications::-webkit-scrollbar {
    height: 6px;
}

.publications::-webkit-scrollbar-track {
    background-color: rgba(64, 67, 98, 0.1);
    border-radius: 8px;
}

.publications::-webkit-scrollbar-thumb {
    background-color: rgba(64, 67, 98, 0.4);
    border-radius: 8px;
}

.event {
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: 8px;
}

.event__img {
    width: 156px;
    height: 173px;
    border-radius: 8px;
    object-position: center;
    object-fit: cover;
}

.event__title {
    font-weight: 400;
    font-size: 14px;
    line-height: 18px;
    color: rgba(255, 255, 255, 0.8);
}

.publications__publishingHouse {
    position: absolute;
    top: 20px;
    left: 14px;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 8px 12px;
    background: rgba(59, 61, 81, 0.6);
    border-radius: 8px;
    z-index: 2;
}

.publications__bogumil {
    z-index: 2;
    position: absolute;
    right: -10px;
    top: -35px;
}

.publications__boxWithArrows {
    position: relative;
}

.publications__scroll {
    position: absolute;
    object-fit: cover;
    top: 0;
    height: 173px;
    width: 54px;
    z-index: 1;
    opacity: 0;
}

.arrowRight {
    right: 0;
}

.arrowLeft {
    transform: rotate(180deg);
    left: 0;
}

.arrowShow {
    opacity: 1;
}

@media only screen and (min-width: 833px) {
    .publicationsBox {
        width: 467px;
        padding: 90px 16px 0px 16px;
        border-radius: 16px;
        background: rgba(81, 81, 81, 0.2);
    }

    .publications {
        width: 433px;
    }

    .event {
        width: 179px;
    }

    .event__img {
        width: 179px;
        height: 213px;
        gap: 8px;
    }

    .publications__bogumil {
        height: 230px;
        width: 190px;
        right: -49px;
    }

    .publications__publishingHouse {
        left: 16px;
        top: 20px;
    }

    .publications__publishingHouse-img {
        width: 195px;
        height: 30px;
    }

    .event__title {
        margin: 0;
    }

    .publications__scroll {
        height: 213px;
    }
}

@media only screen and (min-width: 1439px) {
    .publications {
        column-gap: 15px;
    }

    .event {
        width: 204px;
    }

    .event__img {
        width: 204px;
        height: 257px;
    }

    .event__title {
        font-weight: 400;
        font-size: 16px;
        line-height: 20px;
        margin-bottom: 0px;
    }

    .publications__bogumil {
        height: 255px;
        width: 215px;
    }

    .publications__scroll {
        height: 257px;
    }
}