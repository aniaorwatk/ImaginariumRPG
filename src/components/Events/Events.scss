@import "./../scss/Main";

.publications {
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    overflow: hidden;
    padding: 80px 14px 0 14px;
    border-radius: 8px;
    background-color: $color-dark-trans03;
    filter: $color-filter-publications;

    &__bogumil {
        position: absolute;
        right: -10px;
        top: -35px;
        z-index: 2;
    }

    &__publishingHouse {
        position: absolute;
        top: 20px;
        left: 14px;
        display: flex;
        justify-content: center;
        align-items: center;
        padding: 8px 12px;
        background: $color-bg-publishingHouse;
        border-radius: 8px;
        z-index: 2;
    }

    &__boxWithArrows {
        position: relative;

        &-scroll {
            position: absolute;
            object-fit: cover;
            top: 0;
            height: 213px;
            width: 0px;
            z-index: 1;
        }

        .arrowRight {
            right: 0;
        }

        .arrowLeft {
            transform: rotate(180deg);
            left: 0;
        }

        .arrowShow {
            width: 30px;
        }

        &-events {
            display: flex;
            flex-wrap: nowrap;
            column-gap: 20px;
            width: 317px;
            overflow-y: hidden;

            &::-webkit-scrollbar {
                height: 6px;
            }

            &::-webkit-scrollbar-track {
                background-color: $color-scrollbar-track;
                border-radius: 8px;
            }

            &::-webkit-scrollbar-thumb {
                background-color: $color-scrollbar-thumb;
                border-radius: 8px;
            }
        }
    }
}

.publication {
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: 8px;

    &__img {
        width: 179px;
        height: 213px;
        border-radius: 8px;
        object-position: center;
        object-fit: cover;
    }

    &__title {
        font-weight: 400;
        font-size: 14px;
        line-height: 18px;
        color: $color-white-trans08;
    }
}

@media only screen and (min-width: 833px) {
    .publications {
        padding: 90px 16px 0px 16px;
        border-radius: 16px;
        background: $color-bg-publicationsM;

        &__bogumil {
            height: 230px;
            width: 190px;
            right: -49px;
        }

        &__publishingHouse {
            left: 16px;
            top: 20px;

            &-img {
                width: 195px;
                height: 30px;
            }
        }

        &__boxWithArrows {
            &-events {
                width: 433px;
            }

            .arrowShow {
                width: 54px;
            }
        }
    }

    .publication {
        width: 179px;

        &__img {
            gap: 8px;

            &:hover {
                opacity: 0.85;
            }
        }

        &__title {
            margin: 0;
        }
    }
}

@media only screen and (min-width: 1439px) {
    .publications {
        &__bogumil {
            height: 255px;
            width: 215px;
        }

        &__boxWithArrows {
            &-events {
                column-gap: 15px;
            }
        }
    }

    .publication {
        width: 204px;

        &__title {
            font-weight: 400;
            font-size: 16px;
            line-height: 20px;
            margin-bottom: 0px;
        }
    }
}
