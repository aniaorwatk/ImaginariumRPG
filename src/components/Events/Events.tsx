import { useEffect, useState } from "react";
import { getAction, IEventType, IDataType, URL_IMAGINARIUM } from "../../API/APIGet";
import { LogoBLotr } from "../Logo/LogoBLotr";
import { LogoPublishingHouse } from "../Logo/LogoPublishingHouse";
import "./Events.css"

const Events = () => {
    const [events, setEvents] = useState<IEventType[]>([])

    useEffect(() => {
        getAction(URL_IMAGINARIUM, (data: IDataType) => {
            setEvents(data.events)
        })
    }, [events]);

    const allEvents = events.map(event => {
        return (
            <div key={event.id} className="event">
                <img src={event.filename} alt={event.title} className="event__img" />
                <p className="event__title">{event.title}</p >
            </div>
        )
    })

    return (
        events &&
        <div className="eventsBox" >
            <LogoBLotr className="events__bogumil" />
            <div className="events__publishingHouse">
                <LogoPublishingHouse className="events__publishingHouse-img" />
            </div>
            <div className="events">
                {allEvents}
            </div>
        </div>
    )
}

export default Events
