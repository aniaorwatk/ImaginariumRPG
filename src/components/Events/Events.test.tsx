import { screen, render } from "@testing-library/react";
import { unmountComponentAtNode } from "react-dom";
import Events from "./Events";

describe('Events Component', () => {
    let container: any = null;

    beforeEach(() => {
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });

    it('Should render', async () => {
        const { asFragment } = render(<Events newPublication={undefined} events={undefined} />, container);

        await screen.findAllByText('Title 1');

        expect(asFragment()).toMatchSnapshot();
    })

    it('Should display have two events', async () => {
        render(<Events newPublication={{filename: "banner.jpg", alt: "Title 1", title:"Title 1"}} events={undefined} />, container);

        const event1 = await screen.findAllByText('Title 1');
        const event2 = await screen.findAllByText('Title 2');

        expect(event1.length).toBe(1);
        expect(event2.length).toBe(1);
    })
})
