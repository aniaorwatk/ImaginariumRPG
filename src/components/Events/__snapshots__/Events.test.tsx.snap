// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events Should render 1`] = `
<DocumentFragment>
  <div
    class="events"
  >
    <svg
      class="events__bogumil"
    >
      bLotr.svg
    </svg>
    <div
      class="events__publishingHouse"
    >
      <svg
        class="events__publishingHouse-img"
      >
        wydawnictwo.svg
      </svg>
    </div>
    <div
      class="event"
    >
      <img
        alt="Title 1"
        class="event__img"
        src="test1.jpg"
      />
      <p
        class="event__title"
      >
        Title 1
      </p>
    </div>
    <div
      class="event"
    >
      <img
        alt="Title 2"
        class="event__img"
        src="test2.jpg"
      />
      <p
        class="event__title"
      >
        Title 2
      </p>
    </div>
  </div>
</DocumentFragment>
`;
